api/src/test/java/org/openmrs/module/fhirExtension/translators/DiagnosticReportObsTranslatorHelperTest.java

```diff
@@ -0,0 +1,216 @@
+
+        assertPatientAndObservationDate(testDate, obsModel);
+        assertEquals(testConcept, obsModel.getConcept());
+        Set<Obs> obsGroupMembersTopLevel = obsModel.getGroupMembers();
+        assertEquals(1, obsGroupMembersTopLevel.size());
+
+        Obs obsModelSecondLevel = obsGroupMembersTopLevel.iterator().next();
+        assertPatientAndObservationDate(testDate, obsModelSecondLevel);
+        assertEquals(testConcept, obsModelSecondLevel.getConcept());
+        Set<Obs> obsGroupMembersSecondLevel = obsModelSecondLevel.getGroupMembers();
+        assertEquals(3, obsGroupMembersSecondLevel.size());
+
+        obsGroupMembersSecondLevel.forEach(obsModelChildLevel -> {
+            assertPatientAndObservationDate(testDate, obsModelChildLevel);
+            String obsChildConceptName = obsModelChildLevel.getConcept().getPreferredName(ENGLISH).getName();
+            switch (obsChildConceptName) {
+                case "LAB_REPORT":
+                    assertEquals(REPORT_URL, obsModelChildLevel.getValueText());
+                    break;
+                case "LAB_NOTES":
+                    assertEquals(LAB_TEST_NOTES, obsModelChildLevel.getValueText());
+                    break;
+                case "LAB_RESULT":
+                    assertEquals(REPORT_NAME, obsModelChildLevel.getValueText());
+                    break;
+            }
+        });
+    }
+	
+	@Test
+	public void shouldTranslateToObsWhenOnlyLabNotesAreAvailableForSingleTestWithoutAnyOrder() {
+		Concept testConcept = new Concept();
+		Patient patient = new Patient();
+		patient.setId(PATIENT_ID);
+		Date testDate = new Date();
+		LabResult labResult = new LabResult();
+		labResult.setPatient(patient);
+		labResult.setObservationDate(testDate);
+		labResult.setConcept(testConcept);
+		labResult.setLabReportNotes(LAB_TEST_NOTES);
+		mockConceptServiceGetConceptByName();
+		
+		Obs obsModel = diagnosticReportObsResultTranslatorHelper.createObs(labResult);
+		
+		assertPatientAndObservationDate(testDate, obsModel);
+		assertEquals(testConcept, obsModel.getConcept());
+		Set<Obs> obsGroupMembersTopLevel = obsModel.getGroupMembers();
+		assertEquals(1, obsGroupMembersTopLevel.size());
+		
+		Obs obsModelSecondLevel = obsGroupMembersTopLevel.iterator().next();
+		assertPatientAndObservationDate(testDate, obsModelSecondLevel);
+		assertEquals(testConcept, obsModelSecondLevel.getConcept());
+		Set<Obs> obsGroupMembersSecondLevel = obsModelSecondLevel.getGroupMembers();
+		assertEquals(1, obsGroupMembersSecondLevel.size());
+		
+		Obs obsModelThirdLevel = obsGroupMembersSecondLevel.iterator().next();
+		assertPatientAndObservationDate(testDate, obsModelSecondLevel);
+		assertEquals("LAB_NOTES", obsModelThirdLevel.getConcept().getPreferredName(ENGLISH).getName());
+		assertEquals(LAB_TEST_NOTES, obsModelThirdLevel.getValueText());
+	}
+	
+	@Test
+    public void shouldTranslateToObsWhenTestReportIsUploadedWithLabNotesForPanelWithoutAnyOrder() {
```

Comment:
does it matter if the panel comes from order or not?

Reply:
At the moment, no. Nothing unique in the payload if it comes from order.

Reply:
Then you should consider removing it from your test name

Reply:
Sure. Will make the change accordingly.